admin_guide/cluster_capacity.adoc

```diff
@@ -0,0 +1,184 @@
+about how many pods can be scheduled on each node in the cluster. An example of verbose output on
+a 2-node cluster is as follows:
+
+```
+small-pod pod requirements:
+	- CPU: 150m
+	- Memory: 100Mi
+
+The cluster can schedule 52 instance(s) of the pod small-pod.
+
+Termination reason: Unschedulable: No nodes are available that match all of the following predicates:: Insufficient cpu (2).
+
+Pod distribution among nodes:
+small-pod
+	- 192.168.124.214: 26 instance(s)
+	- 192.168.124.120: 26 instance(s)
+
+```
+
+[admin-guide-cluster-capacity-pod]
+### Running cluster capacity in a pod
+
+Below is an example of a pod spec (`cluster-capacity-pod.yaml`) that runs cluster capacity image as a container. Note that
+input pod spec (`pod.yaml`) to cluster capacity analysis is mounted in a volume using `hostPath`.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: cluster-capacity
+  labels:
+    name: cluster-capacity
+spec:
+  containers:
+  - name: cluster-capacity
+    image: openshift/origin-cluster-capacity
+    env:
+    - name: CC_INCLUSTER
+      value: "true"
+    command:
+    - "/bin/sh"
+    - "-ec"
+    - |
+      /bin/cluster-capacity --podspec=/test-pod/pod.yaml
+    volumeMounts:
+    - mountPath: /test-pod
+      name: test-volume
+  serviceAccountName: cluster-capacity-sa
+  volumes:
+  - name: test-volume
+    hostPath:
+      # directory location on host
+      path: /data
+```
+
+Also note that the above pod spec specifies the following environment variable which is required to
+let the cluster capacity command know that it is running inside a cluster as a pod.
+
+```yaml
+env:
+- name: CC_INCLUSTER
+  value: "true"
+```
+
+By default, this pod would be created in the `default` namespace by using `default` service account. However, the above
+pod spec refers to a different service account (`cluster-capacity-sa`) in the `default` namespace created as below:
+
+```
+$ oc create sa cluster-capacity-sa
+```
+
+This service account `cluster-capacity-sa` is assigned `system:kube-scheduler` cluster scope role below, as the cluster capacity command
+simulates the default scheduler in kubernetes.
+
+```shell
+$ oadm policy add-cluster-role-to-user system:kube-scheduler system:serviceaccount:default:cluster-capacity-sa
+```
+
+However, the cluster role `system:kube-scheduler` does not have enough permissions to run the
+cluster capacity command, as the command accesses other resources
+("limitranges", "serviceaccounts", "secrets", "namespaces", "resourcequotas) too, not part of `system:kube-scheudler` role.
+To satisfy access to these
+resources, a new cluster role (`kube-scheduler-reader-plus`) is created as below:
+
+```shell
+$ cat << EOF| oc create -f -
+kind: ClusterRole
+apiVersion: v1
+metadata:
+  name: kube-scheduler-reader-plus
+rules:
+- apiGroups: [""]
+  resources: ["limitranges", "serviceaccounts", "secrets", "namespaces", "resourcequotas"]
```

Comment:
After adding kube-scheduler role to the cluster capacity sa, these resources were throwing error and required to be added when running cluster capacity, so I added them.